ops/model.py

```diff
@@ -272,6 +274,9 @@ def __init__(self, name, backend, cache):
         self._is_our_unit = self.name == self._backend.unit_name
         self._status = None
 
+        if self._is_our_unit:
+            self._containers = {name: Container(name) for name in cache._meta.containers}
```

Comment:
If we do that it means we cannot ever make this not a dictionary, as it will break compatibility with existing clients. I think we've used that rationale to move them towards a richer type, which enables us to implement richer functionality around the type (such as lazy loading).

Should we do the same here?

Reply:
Fixed. I've added a `ContainerMapping` type for this. Thought about using the existing `LazyMapping`, but it's not really lazy so just added a simple dedicated type. Also means it's not mutable, which I think is a good thing.